### DISCUSSION TITLE
Nullable should warn if a struct has an array, due to it defaulting to null

### DISCUSSION THREAD

**Discussion by jklw10** [dotnet/csharplang#5870] [Language Ideas]
![image](https://user-images.githubusercontent.com/8884828/156747086-cdd52773-22ca-42e9-b316-c56e492962a8.png)

In this case the compiler doesn't have a problem with trying to get a number from Foo.c[]  even though it will be a null dereference because the array will default to being a null.

**Comment by ufcpp** [comment]
See https://github.com/dotnet/csharplang/discussions/5337

**Reply by jklw10** [reply]
Similar, but all I'm asking for is that there's a warning when trying to access nullable types from struct arrays/ other places where default is used.

**Reply by HaloFour** [reply]
IMO the frequency at which those fields would be exposed from the struct directly would be in the minority and any property accessors would be effectively opaque, which would render another layer of flow analysis quite expensive for probably very little real gain.
